plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/GitEventsHandler.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.ext.git.client;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.HashSet;
+import java.util.Set;
+import org.eclipse.che.api.core.jsonrpc.commons.RequestHandlerConfigurator;
+import org.eclipse.che.api.git.shared.FileChangedEventDto;
+import org.eclipse.che.api.git.shared.RepositoryDeletedEventDto;
+import org.eclipse.che.api.git.shared.RepositoryInitializedEventDto;
+import org.eclipse.che.api.git.shared.StatusChangedEventDto;
+import org.eclipse.che.api.project.shared.dto.event.GitCheckoutEventDto;
+
+/**
+ * Receives git events from server side.
+ *
+ * <p>To avoid a risk of IDE slow down this handler bypasses event bus because of heavy load.
+ *
+ * @author Igor Vinokur
+ * @author Mykola Morhun
+ */
+@Singleton
+public class GitEventsHandler implements GitEventSubscribable {
+
+  private static final String EVENT_GIT_FILE_CHANGED = "event/git-change";
+  private static final String EVENT_GIT_STATUS_CHANGED = "event/git/status-changed";
+  private static final String EVENT_GIT_CHECKOUT = "event/git-checkout";
```

Comment:
it would be nice to have these constants shared between server and client code

Reply:
Yes, it would be great. But the issue is common for all rpc channels and probably we have to find general solution.